yt_dlp/extractor/livestream.py

```diff
@@ -207,21 +223,39 @@ def _extract_event(self, event_data):
                     'http://livestream.com/accounts/%s/events/%s/videos/%s' % (account_id, event_id, v_id),
                     'Livestream', v_id, v.get('caption')))
             last_video = videos_info[-1]['id']
+        return entries
+
+    def _extract_event(self, event_data):
+        event_id = compat_str(event_data['id'])
+        entries = self._generate_event_playlist(event_data)
         return self.playlist_result(entries, event_id, event_data['full_name'])
 
+    def _extract_account_events(self, account_data):
+        upcoming_events = account_data['upcoming_events']
+        past_events = account_data['past_events']
+        entries = []
+        events = upcoming_events['data'] + past_events['data']
+        for event_data in events:
+            playlist = self._generate_event_playlist(event_data)
+            if playlist:
+                entries = entries + playlist
+        return self.playlist_result(entries, account_data['id'], account_data['full_name'])
+
     def _real_extract(self, url):
         mobj = self._match_valid_url(url)
         video_id = mobj.group('id')
         event = mobj.group('event_id') or mobj.group('event_name')
         account = mobj.group('account_id') or mobj.group('account_name')
```

Comment:
This needs a rewrite, but I won't insuiste coz it's legacy code

Reply:
@pukkandan I'll do the rewrite, please point me to a reference elsewhere in the code. I'll follow the same pattern.